### ISSUE TITLE
Amount decimals

### ISSUE THREAD

**Issue by fpelliccioni** [bitjson/cashtokens#68]
In other protocols (ERC-20, for example) it is common when creating a token, to be able to define the amount of decimal digits to represent the amounts to the user.
How do we achieve the same with CashTokens?

**Comment by A60AB5450353F40E** [comment]
On consensus layer there's no concept of a decimal, we have atomic units, same like we have satoshis for BCH, and the decimal point is a matter of convention and downstream specifications. Jason is working on a [metadata proposal](https://bitcoincashresearch.org/t/higher-level-token-standards-using-cashtokens/912/2), and I have outlined one approach as well, here: https://github.com/bitjson/cashtokens/issues/50#issuecomment-1240449719

**Comment by A60AB5450353F40E** [comment]
>What do you mean by "consensus layer"?

Rules that nodes will enforce to validate TX-es that have token UTXOs in them.

>Yeah, I know, but it is important for the end user and...

You're absolutely right, I'm just not sure where we will define a standard for defining basic metadata like ticker/name/decimals, it will likely be some other CHIP, but at the end we have to come up with a nice standard!

**Comment by cculianu** [comment]
@fpelliccioni i think what they had in mind is that a convention could be devised, either on-chain or off-chain to retrieve such information. This information is not consensus-specific but rather more of a presentation layer issue.

So for example one can imagine a scheme where your genesis txn has an OP-RETURN blob that instructs wallets how many decimal points to use in the UI and what the UI display name is, etc.

Or one can just hit a web service to retrieve this information (requires some trust but no more than how one just add tokens to a metamask wallet)

I think that was the idea. 

Basically this is ui info that doesn’t need to live n consensus, was the thinking

**Comment by bitjson** [comment]
Exactly 👍

There's a Bitcoin Cash Research topic for this here: [Higher-level token standards using CashTokens](https://bitcoincashresearch.org/t/higher-level-token-standards-using-cashtokens/912/2?u=bitjson) (same link @A60AB5450353F40E posted)

> In short, I’m working on a Bitcoin Cash version of [Token Lists](https://tokenlists.org/) (from the Ethereum ecosystem). This has proven to be a fairly successful strategy for distributing and authenticating metadata about tokens, and it is both simple and reliable. There’s even a [multi-chain standard 2](https://github.com/Uniswap/token-list-bridge-utils) aimed at EVM-based coins. While we’re not EVM-based, matching our own standard closely with these existing systems should make implementation much easier for wallets – especially for wallets that already support tokens on other chains.

E.g. here's Gemini's ETH token list, you can see how most tokens use a `decimals` value of `18`, but some use `8`: https://www.gemini.com/uniswap/manifest.json. There are ways of broadcasting this sort of metadata on-chain, but it's ultimately up to wallets to honor it or not. (Some wallets may choose to re-denominate some tokens, even if token issuers want another value.)

**Comment by bitjson** [comment]
RE: metadata – I'm finally able to share a more complete proposal! https://github.com/bitjson/chip-bcmr

And a video talk describing how it works (and some discussion about how it outperforms strategies used in other cryptocurrency ecosystems – even BCH light clients can validate metadata):

[![video](https://user-images.githubusercontent.com/904007/199376296-29b77382-5658-4df1-add6-40cadc83249a.png)](https://www.youtube.com/watch?v=gPOCj0KulNg)

I think this is answered, but feel free to re-open if there's more to discuss here!
